### PR TITLE
Geopy-1400

### PR DESCRIPTION
**Pull request by domfournier**

### CODE REVIEW

File: docs/source/edge_detection.rst [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+from gridded data in a semi-automated fashion. The application uses
+machine vision algorithms from the
+`Scikit-Image <https://scikit-image.org/>`__ open-source package. Edges are exported to `Geoscience ANALYST <https://mirageoscience.com/mining-industry-software/geoscience-analyst/>`__
+for viewing and editing.
+
+.. figure:: ./images/edge_detection/edge_detection_landing.png
+            :align: center
+            :width: 100%
+
+New user? Visit the `Getting Started <getting_started>`_ page.
```
Comment (benk-mira) [inline docs/source/edge_detection.rst:16]: Link is broken

File: docs/source/edge_detection.rst [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+Detection Parameters
+^^^^^^^^^^^^^^^^^^^^
+
+`Scikit-Image.feature.Canny <https://scikit-image.org/docs/dev/auto_examples/edges/plot_canny.html#sphx-glr-auto-examples-edges-plot-canny-py>`__
+ - **Sigma**: Standard deviation of the Gaussian filter used to smooth the input data. Increase the parameter to detect fewer edges in noisy data.
+
+`Scikit-Image.transform.probabilistic_hough_line <https://scikit-image.org/docs/dev/api/skimage.transform.html#probabilistic-hough-line>`__
+ - **Line length**: Filter for the minimum length (pixels) of detected lines. Increase the parameter to extract longer lines.
+ - **Line gap**: Maximum gap between pixels to still form a line.
+    Increase the parameter to create the number of segments that can be detected from disconnected edges.
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply docs/source/edge_detection.rst:43]: I find this confusing.  Took a stab at clearing it up, but will defer to you.

File: docs/source/edge_detection.rst [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+3. The full grid is sub-divided into overlapping **square tiles** defined by
+   the `window size <#Window-size>`__ parameter. Tiling is used to speed
+   up computations but to also avoid skewing the detection of lines along the longest axis of the input grid.
+
+4. For each tile, edges are converted to a line parametric form using
+   the `Hough Line Transform <#Hough-Line-Parameters>`__.
+
+[Optional] The resulting vertices defining the segments can be filtered to reduce the number of unique lines. Collocated
+vertices are merged if they are within a specified distance of each other. This is controlled by the `merge length <#Merge-Length>`__ parameter.
```
Comment (benk-mira) [inline docs/source/edge_detection.rst:88]: link didn't work

File: docs/source/edge_detection.rst [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+     - With 24 pixels line length
+
+   * - .. figure:: ./images/edge_detection/example_result_default.png
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_result_line24.png
+            :align: center
+            :width: 300
+
+Note that longer lines are extracted when increasing the ``line length`` parameter.
```
Comment (benk-mira) [inline docs/source/edge_detection.rst:171]:
```suggestion
Note that shorter lines are filtered when increasing the ``line length`` parameter.
```

File: docs/source/edge_detection.rst [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+Line Gap
+^^^^^^^^
+
+The following figure shows the effect of increasing the ``line gap`` parameter in the Hough Line Transform. For ``line gap=10`` we obtain:
```
Comment (benk-mira) [inline docs/source/edge_detection.rst:176]:
```suggestion
The following figure shows the effect of increasing the ``line gap`` parameter in the Hough Line Transform. For ``line gap=1`` and ``line gap=10`` we obtain:
```

File: docs/source/edge_detection.rst [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - Detection parameters
+     - With ``threshold=1``
+     - With ``threshold=8``
+
+   * - .. figure:: ./images/edge_detection/edge_detection_threshold.png
```
Comment (benk-mira) [inline docs/source/edge_detection.rst:208]: Maybe put the parameters image above to give the side-by-side more space (hard to see).

File: docs/source/trend_lines.rst [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+===========
+
+With this application, users can create trend lines from vertices and data.
+
+.. figure:: ./images/trend_lines/trend_intro.png
+            :align: center
+            :width: 100%
+
+
+New user? Visit the `Getting Started <getting_started>`_ page.
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:13]: Broken link

File: docs/source/trend_lines.rst [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+ - **Data**: [Optional] Referenced data defining properties to link with segments. If omitted, the segments will be created from vertices alone.
+
+.. _filters:
+
+Connection Filters
+^^^^^^^^^^^^^^^^^^
+ - **Parts**: [Optional] Referenced data defining the unique parts of the object. Connections between vertices belonging to the same part are ignored.
+ - **Minimum edges**: Minimum number of segments for a trend line to be valid.
+ - **Maximum distance**: Maximum distance between two points to consider for trend line detection.
+ - **Damping factor**: Threshold parameter [0, 1] penalizing changes in direction between connected segments.
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:38]: Weird indentation

File: docs/source/trend_lines.rst [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        :align: center
+        :width: 300
+
+        This process is repeated recursively until there is no valid neighbouring edge to form a connection with. The metric used to evaluate the tortuosity is expressed as follows:
+
+    .. math::
+
+        \delta = \theta^{(1-\alpha)} * L
+
+    where the angle :math:`\theta` is the change in direction between the incoming and outgoing segments, :math:`L` is the length of the outgoing segment, and :math:`\alpha` is the `Damping factor <filters>`_.
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:82]: Broken link

File: docs/source/trend_lines.rst [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+Methodology
+-----------
+
+The conversion from scatter point data to trend lines is achieved by the following steps:
+
+1. The selected ``Points`` or ``Curve`` object's vertices are triangulated to form a Delaunay mesh.
+   The mesh is used to identify the initial edges connecting the vertices.
+
+    .. figure:: ./images/trend_lines/trend_step1.png
+            :align: center
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:56]: The images are a nice touch! what is the significance of the blue and red dashed lines?

File: docs/source/trend_lines.rst [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - Input data
+     - Default parameters
+     - Result
+
+   * - .. figure:: ./images/trend_lines/example_data.png
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:101]: I would give the images more space by organizing vertically

File: docs/source/trend_lines.rst [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+    where the angle :math:`\theta` is the change in direction between the incoming and outgoing segments, :math:`L` is the length of the outgoing segment, and :math:`\alpha` is the `Damping factor <filters>`_.
+    The path is updated with the connection with the lowest :math:`\delta` value.  Note that for
+    :math:`\alpha=1`, only the length of the outgoing segments are considered. This generally increases the number of connections on a trend line, but the path may be more erratic.
+
+
+Example
+-------
+
+This section demonstrates the effect of various detection parameters on a training dataset provided `here <https://github.com/MiraGeoscience/curve-apps/tree/main/edge_detection-assets>`__.
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:90]: Point to release branch that will contain the asset described (empty on main until we merge)

File: docs/source/trend_lines.rst [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+connecting points that are too far apart or allowed to change direction rapidly. We can improve the results by adjusting the parameters.
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - Detection parameters
+     - Result
+
+   * - .. figure:: ./images/trend_lines/example_filtered_ui.png
```
Comment (benk-mira) [inline docs/source/trend_lines.rst:121]: I would align these vertically too, with last sentence in between the two images.

File: docs/source/getting_started.rst [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+    :width: 200
+
+Registering the Conda distribution as the default Python 3.10 interpreter is totally optional.
+Preferably uncheck that box if you already have Python 3 installed on your system.
+
+.. note:: We recommend installing **Miniforge**: beyond being smaller,
+    it also installs packages from the conda-forge repository by default,
+    which has no restriction for commercial use, while both Miniconda and Anaconda distributions use
+    the Anaconda repository by default: see `Anaconda Terms of Service <https://www.anaconda.com/terms-of-service>`_ for details.
+    In any case, the installation of **octree-creation** forces the usage of the conda-forge repository,
```
Comment (benk-mira) [inline docs/source/getting_started.rst:38]:
```suggestion
    In any case, the installation of **curve-apps** forces the usage of the conda-forge repository,
```

File: docs/source/getting_started.rst [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+    and is thus not affected by the Anaconda Terms of Service.
+
+Download the latest octree-creation
```
Comment (benk-mira) [inline docs/source/getting_started.rst:41]:
```suggestion
Download the latest curve-apps
```

File: docs/source/getting_started.rst [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+.. figure:: /images/getting_started/extract.png
+    :align: center
+    :width: 50%
+
+
+Run ``Install_or_Update.bat``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The same batch file can be used to install or update **octree-creation**.
```
Comment (benk-mira) [inline docs/source/getting_started.rst:59]:
```suggestion
The same batch file can be used to install or update **curve-apps**.
```

**Review by benk-mira — CHANGES_REQUESTED** [review]
Just noticed these on my docs pr..

File: docs/source/edge_detection.rst (hunk 1 above)
Comment (domfournier) [reply docs/source/edge_detection.rst:16]: These won't work until we are fully build

File: docs/source/edge_detection.rst [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+ - **Threshold**: Threshold parameter used in the Hough Line Transform.
+
+
+**[Optionals]**
+
+ - **Window size**: Size of the square window used to sub-divide the grid for processing. By default, the window size
+    is set to the shortest side of the input grid. Smaller window sizes can be used to speed up computations but may result in
+    more fragmented lines. Larger window sizes can be used to improve line continuity but may slow down computations.
+ - **Merge length**: Merge lines within a specified distance (in meters) of each other. This parameter is useful for
+    merging fragmented lines that are close to each other but not connected.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/edge_detection.rst:53]: Line gap is a detection parameter on pixels, this one merges the resulting vertices to reduce complexity. I re-worded a bit.

File: docs/source/edge_detection.rst [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+ - **Save as**: Assign a specific name to the resulting ``Curve`` object.
+ - **Group**: Create a ``Container Group`` entity to store the results.
+
+
+.. _edge_methodology:
+
+Methodology
+-----------
+
+The conversion from raster data to lines involves the following four
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/edge_detection.rst:68]: I get your point. Curves is the final product, made up of many "segments".

File: docs/source/edge_detection.rst (hunk 3 above)
Comment (domfournier) [reply docs/source/edge_detection.rst:88]: Fixed

File: docs/source/edge_detection.rst (hunk 6 above)
Comment (domfournier) [reply docs/source/edge_detection.rst:208]: Done

File: docs/source/trend_lines.rst (hunk 8 above)
Comment (domfournier) [reply docs/source/trend_lines.rst:38]: Shifted

File: docs/source/trend_lines.rst (hunk 10 above)
Comment (domfournier) [reply docs/source/trend_lines.rst:56]: Re-edited for clarity

File: docs/source/trend_lines.rst (hunk 12 above)
Comment (domfournier) [reply docs/source/trend_lines.rst:90]: I would rather point to the stable version. Ok if broken at first

File: docs/source/edge_detection.rst [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+[Optional] The resulting vertices defining the segments can be filtered to reduce the number of unique lines. Collocated
+vertices are merged if they are within a specified distance of each other. This is controlled by the `merge length <#Merge-Length>`__ parameter.
+
+The final output is a ``Curve`` object containing the detected lineaments.
+The orientation of the lines, in absolute degree from North [0, 90], and line length are stored as attributes of the ``Curve`` object.
+
+Example
+-------
+
+This section demonstrates the effect of various detection parameters on a training grid provided `here <https://github.com/MiraGeoscience/curve-apps/tree/main/edge_detection-assets>`__.
```
[the remark this answers is not in the document]
Comment (domfournier) [reply docs/source/edge_detection.rst:94]: I would rather not to have to edit those paths again. We can have them broken at first

**Review by benk-mira — APPROVED** [review]
Better still!

**Pull request merged**